Content.Server/Chemistry/EntitySystems/InjectorSystem.cs

```diff
@@ -330,8 +331,17 @@ private void AfterDraw(Entity<InjectorComponent> injector, EntityUid target)
             return;
         }
 
+        var applicableTargetSolution = targetSolution.Comp.Solution;
+        // If a whitelist exists, remove all non-whitelisted reagents from the target solution temporarily
+        var temporarilyRemovedSolution = new Solution();
+        if (injector.Comp.ReagentWhitelist is { } reagentWhitelist)
+        {
+            var reagentPrototypeWhitelistArray = reagentWhitelist.Select(x => (string) x).ToArray();
+            temporarilyRemovedSolution = applicableTargetSolution.SplitSolutionWithout(applicableTargetSolution.Volume, reagentPrototypeWhitelistArray);
```

Comment:
Can you do this without the LINQ as that will allocate a lot.

Reply:
Does LINQ just suck and I should never use functions from it?

Reply:
> Does LINQ just suck and I should never use functions from it?

yes